src/Control/Carrier/NonDet/Church.hs

```diff
@@ -21,7 +31,13 @@ import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
 
-runNonDet :: (m b -> m b -> m b) -> (a -> m b) -> m b -> NonDetC m a -> m b
+-- | Run a 'NonDet' effect, using the provided functions to interpret
+-- choice, leaf results, and failure.
+runNonDet :: (m b -> m b -> m b) -- ^ Handles choice ('Control.Effect.Choose.<|>')
+          -> (a -> m b)          -- ^ Handles embedding results ('pure')
+          -> m b                 -- ^ Handles failure ('Control.Effect.Empty.empty')
+          -> NonDetC m a         -- ^ A nondeterministic
+          -> m b
```

Comment:
If we’re going to reformat this, can we please format it so that the `::` goes on the next line from the function name, and everything is indented only two spaces?